### PR TITLE
Fixed write file error in examples; encode to UTF-8 (#5306

### PR DESCRIPTION
**Pull request by amyxst**
All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [ ] issues: fixes #xxxx
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

### CODE REVIEW

File: examples/embed/embed_multiple.py [hunk 1]
```diff
@@ -77,6 +77,6 @@
                        div=div)
 
 with open(filename, 'w') as f:
-    f.write(html)
+    f.write(html.encode('utf8'))
```
Comment (mattpap) [inline examples/embed/embed_multiple.py]: A better solution would be to use `io.open(..., encoding='utf-8')`, especially using it uniformly across code base.

**Comment by amyxst** [comment]
Hi Matt,

I apologize this is my first time attempting to contribute to a project.
Would you suggest that I look through the codebase to find other areas
where the encoding may need to be fixed as well?

Best,

Amy

**Comment by bryevdv** [comment]
Hi @amyxst no apologies, thanks for the contribution! So all of the "higher level" output and embed functions like `file_html` or `autoload_server` in Bokeh automatically use the `encode_utf8` function in this module:

https://github.com/bokeh/bokeh/blob/master/bokeh/util/string.py

So I think the only other areas that might need fixing would be any other example that do a `.write` on raw strings that didn't come from `file_html`, etc. It looks like there are are few other *potential* cases:
```
./embed/animated.py:
   63 :     f.write(html)
./embed/embed_multiple.py:
   80 :     f.write(html)
./embed/embed_multiple_responsive.py:
   66 :     f.write(html)
./embed/widget.py:
  177 :     f.write(html)
./howto/interactive_bubble/gapminder.py:
  128 :     f.write(html)
```
But the `html` variables might have from from `file_html`, `autoload_server`, etc. (in which case everything is fine) but it might be worth a quick look at them. 

Regarding the actual PR, I agree with @mattpap that `io.open(..., encoding='utf-8')` might be a better solution, or alternatively you could use the `encode_utf8` function I mentioned like:
```
 f.write(encode_utf8(html))
```
The reason is that there are some differences between py2 and py3 that these ways account for.

**Comment by bryevdv** [comment]
@amyxst just checking in, do you think you will have an opportunity to update the PR in one of the ways described above (using `io,open` or `encode_utf8`)?

**Comment by bryevdv** [comment]
Hi @amyxst I would like to get this fix in for `0.12.4` but I do think that the solution either needs to use `io.open` as @mattpap described, or the `bokeh.util.string.encode_utf8` as I described. I appreciate that you may not have more time to devote to this, but I'd like to give you an opportunity to close this PR, if you are able. Otherwise this is just a heads up that I will close this PR next week and submit a PR with one of the fixes described above.

**Comment by bryevdv** [comment]
superseded by #5535 

@amyxst thanks for doing the initial work on this, I have closed the issue in a separate PR

**Pull request closed without merging**